### PR TITLE
Enable service account impersonation in GoogleCalendar source connector

### PR DESCRIPTION
**Pull request by lhvubtqn**
This enable service account impersonation (in case the user is organization manager & want to use a service account to access memebers' calendars)

## Description

Current implementation of GoogleCalendar source connector only allows calendar owner or invited accounts to access calendar.

In normal case, this is fine. But in an organization, when admins need to access members' calendars to calculate necessary metrics to determine productivity, forcing all members to share their calendar with an account is tricky.

This PR enable service account impersonation (together with domain-wide delegation), so that an authenticated service account can access organization members' calendars.

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

### CODE REVIEW

**Comment by lhvubtqn** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by lhvubtqn** [comment]
@tovbinm can you have a look at this?

File: sources/googlecalendar-source/src/googlecalendar.ts [hunk 1]
```diff
@@ -59,6 +59,9 @@ export class Googlecalendar {
         private_key: config.private_key.replace(/\\n/g, '\n'),
         client_email: config.client_email,
       },
+      clientOptions: {
+        subject: config.calendar_id
```
Comment (tovbinm) [inline sources/googlecalendar-source/src/googlecalendar.ts:63]: @lhvubtqn Nice!! (assuming this works as expected) Is this being documented somewhere?

Comment (tovbinm) [reply sources/googlecalendar-source/src/googlecalendar.ts:63]: I only found this comment mentioning this - https://github.com/googleapis/google-api-nodejs-client/issues/1834#issuecomment-535773650, but it does not look it many people upvoted it. 

Did you test it yourself?

Comment (lhvubtqn) [reply sources/googlecalendar-source/src/googlecalendar.ts:63]: Yeah, I tested it myself.

As I said, current implementation only allow owners or invited members to ingest calendar data. In order to access to organization's memebers' calendar, I did following steps:

- Grant `domain-wide delegation` for the service account ([follow this](https://developers.google.com/workspace/guides/create-credentials#optional_set_up_domain-wide_delegation_for_a_service_account))
- Edit the code, add `subject` to google auth configurations (Yep, I follow the link to found: [google-api-nodejs-client/issues/1834](https://github.com/googleapis/google-api-nodejs-client/issues/1834#issuecomment-535773650))

Comment (tovbinm) [reply sources/googlecalendar-source/src/googlecalendar.ts:63]: Can you please add a few lines of documentation to Google Calendar source README.md?

I will check what can be done with special characters.

File: sources/googlecalendar-source/README.md [hunk 2]
```diff
@@ -46,6 +46,17 @@ Create credentials:
 - Create [Service Account](https://console.cloud.google.com/apis/credentials)
 - Use client_email and private_key from Service account
 
+#### Enable `domain-wide delegation` for service account (Workspace Admin only)
+
+In normal cases, one's calendar can only be accessed by its owner or specific users that it is shared to. This is good for personal use, but in an organization that required data from all members' calendars, using one service account for each member, or forcing all members to share their calendars with an account is troublesome.
+
+Google has introduced an feature named `domain-wide delegation`, in which it allows an service account to `call APIs on behalf of users in a Google Workspace organization`. With this, one service account can access all organization members' calendars without extra configuration efforts.
+
+Refer to this [link](https://developers.google.com/workspace/guides/create-credentials#optional_set_up_domain-wide_delegation_for_a_service_account) to know how to enable `domain-wide delegation` for the service acccount.
```
Comment (lhvubtqn) [inline sources/googlecalendar-source/README.md:55]: @tovbinm I have added documentation on `README.md` 🥰

File: sources/googlecalendar-source/src/googlecalendar.ts (hunk 1 above)
Comment (tovbinm) [reply sources/googlecalendar-source/src/googlecalendar.ts:63]: The error with special characters is coming from the Google API: `invalid_grant: Invalid email or User ID`. I tried URI encoding it:
```
clientOptions: {
        subject: encodeURI(config.calendar_id),
},
``` 

But it does not seem to work. Need to dig more.

**Comment by tovbinm** [comment]
FYI @lhvubtqn https://github.com/faros-ai/airbyte-connectors/pull/792

**Pull request merged**